core/controlplane/config/templates/cloud-config-controller

```diff
@@ -791,6 +791,119 @@ write_files:
       sed -i -e "s#\$ETCDKEY#$etcd_key#g" /srv/kubernetes/manifests/calico.yaml
 
 {{ end }}
+{{ if .ExportKubeResources }}
+  - path: /srv/kubernetes/manifests/cluster-dump.yaml
+    content: |
+      ---
+      apiVersion: extensions/v1beta1
+      kind: Deployment
+      metadata:
+        name: cluster-dump
+        namespace: kube-system
+        labels:
+          k8s-app: cluster-dump-policy
+      spec:
+        replicas: 1
+        template:
+          metadata:
+            name: cluster-dump
+            namespace: kube-system
+            labels:
+              k8s-app: cluster-dump-policy
+          spec:
+            containers:
+            - name: cluster-dump-dumper
+              image: {{.HyperkubeImage.RepoWithTag}}
+              command: ["/bin/bash", "-c" ]
+              args:
+                - |
+                    set -x ;
+                    DUMP_DIR_COMPLETE=/cluster-dump/complete ;
+                    mkdir -p ${DUMP_DIR_COMPLETE} ;
+                    while true; do
+                      TIMESTAMP=$(date +%Y-%m-%d_%H-%M-%S)
+                      DUMP_DIR=/cluster-dump/tmp/${TIMESTAMP} ;
+                      mkdir -p ${DUMP_DIR} ;
+                      RESOURCES_OUT_NAMESPACE=( namespaces persistentvolumes nodes storageclasses ) ;
```

Comment:
How does it work for `persistentvolumes`? Can you retain the data persisted in a persistent volume after restored from a backup?